### ISSUE TITLE
function calls with multiple return values

### ISSUE THREAD

**Issue by maartenvd** [stevengj/SoftGlobalScope.jl#17]
I have attached an example file as an attachment. I'm not quite sure how to describe the bug other then julia sometimes complaining about undefined variables. I'm using IJulia, so presumably this package is used to make everything global.

https://www.dropbox.com/s/e1crp02stuoixj0/Untitled.ipynb?dl=0

**Comment by stevengj** [comment]
Can you upload it as a gist on github?  That would make it easier to view.   Or just quote the code and the error message in the issue.

**Comment by maartenvd** [comment]
It's a jupyter notebook and I don't know how to use this with gist. I can open a repo and upload it there but before I do this, is the following supposed to work?

```jl
using SoftGlobalScope
a=rand(5)

function dostuff(b)
    return (similar(b),similar(b))
end

@softscope for i in 1:10
    (a,_)=dostuff(a)
end
```

`UndefVarError: a not `

**Comment by stevengj** [comment]
Yes, it would be good to make this work.

**Comment by stevengj** [comment]
A simpler example:
```jl
g = 0
for i = 1:10
    g,_ = (i+g, i+1)
end
```

**Comment by stevengj** [comment]
cc @dawbarton

**Comment by dawbarton** [comment]
The original example and the simpler example now work with #19.

**Comment by stevengj** [comment]
Closed by #19.
